builtin/logical/database/path_config_connection.go

```diff
@@ -292,6 +303,7 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 		delete(data.Raw, "verify_connection")
 		delete(data.Raw, "root_rotation_statements")
 		delete(data.Raw, "password_policy")
+		delete(data.Raw, "disable_escaping")
```

Comment:
I don't think we need to delete `disable_escaping` here, since we end up adding it right back to [ConnectionDetails](https://github.com/hashicorp/vault/pull/13414/files#diff-25b7b056201175f5a21ca0e438bebb95f83648322ec77dd6bde35598e95eb511R326) below. I'd suggest to leave it in `data.Raw` and let [this logic](https://github.com/hashicorp/vault/blob/main/builtin/logical/database/path_config_connection.go#L304-L313) add it to `ConnectionDetails`. After that, you can remove the line on 326, since it will already be added.